lib/src/util.h

```diff
@@ -49,6 +49,10 @@
         name = &_##name; \
     }
 
+#ifdef WIN32
+#undef min
```

Comment:
What is `min` and `max` otherwise defined to on WIN32?

Reply:
From `stdlib.h`

```
#define max(a,b) (((a) > (b)) ? (a) : (b))
#define min(a,b) (((a) < (b)) ? (a) : (b))
```

Reply:
Ok. It might be safer to just `#ifndef WIN32` around the definitions of `min` and `max`, as they are the same. But there again it's nice to have actual functions rather than macros providing that behavior, so perhaps the `undef` approach is good. Your choice.

Reply:
I will keep the undef approach. I prefer the functions to the macros, as they were specifically designed for this lib. Both forms work too.